docs/upgrade.html

```diff
@@ -21,6 +21,10 @@
 0.10.0.0 has <a href="#upgrade_10_breaking">potential breaking changes</a> (please review before upgrading) and
 there may be a <a href="#upgrade_10_performance_impact">performance impact during the upgrade</a>. Because new protocols
 are introduced, it is important to upgrade your Kafka clusters before upgrading your clients.
+<p/>
+<b>Notes to client with version 0.9.0.0: </b>Due to a bug introduced in 0.9.0.0,
+clients with ZooKeeper dependency may not be able to work with 0.10.0.0 broker. Therefore, 0.9.0.0 clients should
```

Comment:
@granthenke, yes, it is a big deal. Your suggestion was discussed in the KIP thread, but @junrao said that we need to increase the version when we change the format. As I understand it, the idea is to move away from being limited to just adding fields once we stop supporting old clients (but that is a while away in my opinion).

Thinking more about this, we could also add a new version field and deprecate the old one (which can stay at `2` until we no longer support `0.9.0.0` many years from now) if we really care about having an incrementing version each time we change the format. Thoughts?